src/yb/master/master-path-handlers.cc

```diff
@@ -1011,8 +1011,14 @@ void MasterPathHandlers::HandleCatalogManager(const Webserver::WebRequest& req,
 
 namespace {
 
-bool CompareByRole(const TabletReplica& a, const TabletReplica& b) {
-  return a.role < b.role;
+bool CompareByHost(const TabletReplica& a, const TabletReplica& b) {
+    TSRegistrationPB reg_a = a.ts_desc->GetRegistration();
+    TSRegistrationPB reg_b = b.ts_desc->GetRegistration();
+    if (!reg_a.common().http_addresses().empty() && !reg_b.common().http_addresses().empty()) {
+        return reg_a.common().http_addresses(0).host() < reg_b.common().http_addresses(0).host();
+    } else {
+        return a.ts_desc->permanent_uuid() < b.ts_desc->permanent_uuid();
```

Comment:
since permanent_uuid is generally expected to not be empty, and since our goal is only to have /some/ stable sorting of hosts, why dont we simplify this logic by always sorting by permanent_uuid?

Reply:
The reason is I was trying to keep the hosts' increasing order if they're IPs. That would be more intuitive I think. But it indeed intriduces a more complex logic here. I could modify it to simply sort by permanent_uuid. What do you think?

Reply:
i think lets prefer simplicity in this case, just to enable quick eyeball comparison of replica balance in 3 node clusters